cromshell

```diff
@@ -0,0 +1,311 @@
+  echo -e "   execution-status-count [worfklow-id] [[worfklow-id]...]"
+  echo -e "   timing [worfklow-id] [[worfklow-id]...]"
+  echo -e "   abort [worfklow-id] [[worfklow-id]...]"
+	echo -e "   notify [workflow-id] email_address"
+  echo 
+  echo -e "Return values:"
+  echo -e "  0                  SUCCESS"
+	echo -e "  ANYTHING_BUT_ZERO  FAILURE/ERROR"
+  echo
+}
+
+#Display a message to std error:
+function error() 
+{
+  echo "$1" 1>&2 
+}
+
+TMPFILELIST=''
+function makeTemp()
+{
+  local f
+  f=$( mktemp )
+  TMPFILELIST="${TMPFILELIST} $f"
+  echo $f
+}
+
+function cleanTempVars()
+{
+  rm -f ${TMPFILELIST}
+}
+
+function at_exit()
+{
+  cleanTempVars
+}
+
+function checkPipeStatus() {
+	for st in ${PIPESTATUS[@]} ; do 
+		[ $st -ne 0 ] && return $st
+	done
+	return 0
+}
+
+################################################################################
+
+trap at_exit EXIT 
+
+################################################################################
+
+#Check given arguments:
+if [[ $# -gt $MAXARGS ]] ; then
+  usage
+  exit 1
+elif [[ $# -lt $MINARGS ]] ; then
+  usage
+  exit 2
+fi
+
+################################################################################
+
+function populateWorkflowId() {
+
+	local userSpecifiedId=$1
+
+	# If the user specified a negative number, get the nth last workflow:
+  if [[ $userSpecifiedId == -* ]]; then
+  	local row=${userSpecifiedId#-}
+  	WORKFLOW_ID=$(tail -n $row $CROMWELL_DIARY_WORKFLOW_FILE | head -n 1)
+  else
+		# If the user specified anything at this point, assume that it 
+		# is a workflow UUID:
+  	if [ -n "$userSpecifiedId" ]; then 
+			WORKFLOW_ID=$userSpecifiedId 
+		# If the user specified nothing, get the last workflow ID:	
+		else 
+			WORKFLOW_ID=$(cat ${CROMWELL_LAST_WORKFLOW_FILE} )
+		fi
+  fi
+}
+
+# Submit a workflow and arguments to the Cromwell Server
+function submit() { 
+  local response=$(curl -s -F workflowSource=@${1}  ${2:+ -F workflowInputs=@${2}} ${3:+ -F workflowOptions=@${3}} ${4:+ -F workflowDependencies=@${4}} ${CROMWELL_URL}/api/workflows/v1)
+	local r=$?
+  echo $response  
+
+	[ $r -ne 0 ] && error "FAILED TO SUBMIT JOB" && return $r
+
+  local id=$(echo $response | cut -d"," -f1 | cut -d":" -f2 | sed s/\"//g | sed s/\ //g)
+  mkdir $id
+  cp ${1} ${2} ${3} ${4} ${id}/
+  echo $id > $CROMWELL_LAST_WORKFLOW_FILE
+  echo $id >> $CROMWELL_DIARY_WORKFLOW_FILE
+	return 0
+}
+
+# Check the status of a Cromwell job UUID
+function status() {
+  curl -s ${CROMWELL_URL}/api/workflows/v1/${1}/status | jq . ;
+	return $( checkPipeStatus )
+}
+
+# Get the logs of a Cromwell job UUID
+function logs() {
+  curl -s ${CROMWELL_URL}/api/workflows/v1/${1}/logs | jq . ;
+	return $( checkPipeStatus )
+}
+
+# Get the metadata for a Cromwell job UUID
+function metadata() {
+  curl --compressed -s ${CROMWELL_URL}/api/workflows/v1/${1}/metadata?${CROMWELL_METADATA_PARAMETERS} | jq . ;
+	return $( checkPipeStatus )
```

Comment:
Would `Set +e` accomplish a similar thing to the pipe status checks?   It seems like it's reporting any non-0 exit code for any commands that were connected to the pipe?

Reply:
Yes, but then you get the `bash` equivalent of a stack trace.

I'll update the check status method to be a little more descriptive.